x/auth/client/cli/tx_sign.go

```diff
@@ -230,64 +226,52 @@ func makeSignCmd() func(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		// if --signature-only is on, then override --append
-		generateSignatureOnly, _ := cmd.Flags().GetBool(flagSigOnly)
-		multisigAddrStr, _ := cmd.Flags().GetString(flagMultisig)
-
-		from, _ := cmd.Flags().GetString(flags.FlagFrom)
+		printSignatureOnly, _ := f.GetBool(flagSigOnly)
+		multisigAddrStr, _ := f.GetString(flagMultisig)
+		from, _ := f.GetString(flags.FlagFrom)
 		_, fromName, err := client.GetFromFields(txFactory.Keybase(), from, clientCtx.GenerateOnly)
 		if err != nil {
 			return fmt.Errorf("error getting account from keybase: %w", err)
 		}
 
+		overwrite, _ := f.GetBool(flagOverwrite)
 		if multisigAddrStr != "" {
 			var multisigAddr sdk.AccAddress
-
 			multisigAddr, err = sdk.AccAddressFromBech32(multisigAddrStr)
 			if err != nil {
 				return err
 			}
-
 			err = authclient.SignTxWithSignerAddress(
-				txF, clientCtx, multisigAddr, fromName, txBuilder, clientCtx.Offline,
-			)
-			generateSignatureOnly = true
+				txF, clientCtx, multisigAddr, fromName, txBuilder, clientCtx.Offline, overwrite)
+			printSignatureOnly = true
```

Comment:
I think the `generateSignatureOnly = true` line (which I think you're attempting to reproduce here) is actually an artifiact of a previous misunderstanding, whereby signing a multisig transaction is expected to set `append=true` and ignore hte `--append` flag, but instead, the logic was implemented incorrectly to ignore the `--sig-only` flag and have `generateSignatureOnly` set to true.

Maybe i'm misunderstanding things, but I think there is a valid use case for a person wanting to sign a multisig transaction and receive back the full Tx (not just the generated signature).

Reply:
This part follows the command description:
```
The --multisig=<multisig_key> flag generates a signature on behalf of a multisig account
key. It implies --signature-only. Full multisig signed transactions may eventually
be generated via the 'multisign' command.
```
Probably the transaction will have to change.  Do you think we should change that behavior?

Reply:
@alessio  - you may have more context here.

Reply:
Ahh i didn't see that command description. Maybe i don't fully understand, so i'll default to @alessio here.

Reply:
>  I think there is a valid use case for a person wanting to sign a multisig transaction and receive back the full Tx (not just the generated signature)

@clevinson 

@robert-zaremba understood correctly. `sign --sig-only` was meant to print signatures only. When you pass a multisig address (via `--multisig=`), the behaviour is rightly the same print-only 'cause you can't really generate a multisig signature in one go [1].

> Do you think we should change that behavior?

IMO the command's behaviour should not change. A signed multisig tx is designed to be signed by multiple parties at different times. When I did the work in the first place, it made sense to me to keep the `sign` and `multisign` commands separate to prevent the `sign` command's logic from growing too big.


[1] In theory yes, you could, provided that you had all required multisig key's keys stored in your keyring. Though I suspect that that would defeat the point of holding assets in a multisig :)